### PR TITLE
CTR needs to support boot property styles for the tablePrefix property

### LINKED ISSUE #5854: Task table prefix is not resolved correctly in CTR
**Description:**
Currently we are receiving a lot of warnings when starting a composed task. During our migration we found out that the lookup of the  table prefix is broken as the relax binding rules of spring are resolving the property `TABLEPREFIX` to `tableprefix` instead of `tablePrefix`

See:

{"timestamp":"2024-06-27T10:17:41.283+0200","level":"WARN","thread":"main","logger":"org.springframework.cloud.dataflow.composedtaskrunner.ComposedTaskRunnerConfiguration","message":"Cannot find app.s3-uploader-app.spring.cloud.task.**tablePrefix** in {...., app.s3-uploader-app.spring.cloud.task.**tableprefix**=BOOT3_TASK_,....}

**Release versions:**
2.11.2

**Custom apps:**
N/A

**Steps to reproduce:**
Just launch a task with the new CTR

**Screenshots:**
N/A

**Additional context:**
N/A

**Solution:**

Replace code:

```java
logger.debug("addTaskExplorer:{}", taskName);
String propertyName = String.format("app.%s.spring.cloud.task.tablePrefix", taskName);
String prefix = properties.getComposedTaskAppProperties().get(propertyName);
if (prefix == null) {
	prefix = env.getProperty(propertyName);
}
if (prefix != null) {
	TaskExecutionDaoFactoryBean factoryBean = new MultiSchemaTaskExecutionDaoFactoryBean(dataSource, prefix);
	logger.debug("taskExplorerContainer:adding:{}:{}", taskName, prefix);
	explorers.put(taskName, new SimpleTaskExplorer(factoryBean));
} else {
	logger.warn("Cannot find {} in {} ", propertyName, properties.getComposedTaskAppProperties());
}
```

With:

```java
logger.debug("addTaskExplorer:{}", taskName);

List<String> propertyNames = Stream.of("app.%s.spring.cloud.task.tablePrefix",
        "app.%s.spring.cloud.task.table-prefix",
        "app.%s.spring.cloud.task.tableprefix")
    .map(propertyName -> String.format(propertyName, taskName)).toList();

String prefix = propertyNames.stream()
    .map(propertyName -> {
        String prefixOfComposedTaskProperties = properties.getComposedTaskAppProperties().get(propertyName);
        return prefixOfComposedTaskProperties == null ? env.getProperty(propertyName) : prefixOfComposedTaskProperties;
    })
    .filter(Objects::nonNull)
    .findFirst().orElse(null);

if (prefix != null) {
	TaskExecutionDaoFactoryBean factoryBean = new MultiSchemaTaskExecutionDaoFactoryBean(dataSource, prefix);
	logger.debug("taskExplorerContainer:adding:{}:{}", taskName, prefix);
	explorers.put(taskName, new SimpleTaskExplorer(factoryBean));
} else {
	logger.warn("Cannot find {} in {} ", propertyNames, properties.getComposedTaskAppProperties());
}
```

in the CTR. You might need to downport it if you don't want to use Java 17. 😁

### PR DESCRIPTION
**Pull request by cppwfs**
See #5854

### CODE REVIEW

**Review by klopfdreh — APPROVED** [review]
Looks good to me 👍

File: spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/ComposedTaskRunnerConfiguration.java [hunk 1]
```diff
@@ -102,17 +107,28 @@ private static void addTaskExplorer(
 			String taskName
 	) {
 		logger.debug("addTaskExplorer:{}", taskName);
-		String propertyName = String.format("app.%s.spring.cloud.task.tablePrefix", taskName);
-		String prefix = properties.getComposedTaskAppProperties().get(propertyName);
-		if (prefix == null) {
-			prefix = env.getProperty(propertyName);
-		}
+		List<String> propertyNames = new ArrayList<>();
+		RelaxedNames relaxedNames = RelaxedNames.forCamelCase("tablePrefix");
```
Comment (corneil) [inline spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/ComposedTaskRunnerConfiguration.java:111]: I would do:
```
RelaxedNames relaxedNames = RelaxedNames.forCamelCase(String.format("app.%s.spring.cloud.task.tablePrefix", taskName));
relaxedNames.forEach(propertyNames::add);
```
This way you are sure in the case of a strange taskName you still match a valid property.
Keep in mind that env.getProperty uses relaxed matching.

Comment (cppwfs) [reply spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/ComposedTaskRunnerConfiguration.java:111]: Unfortunately that will not work.   The `taskName` portion of the property id is also adjusted along with the `tablePrefix` if you pass the entire id into the forCamelCase.  Thus causing a miss when looking up the property.

Comment (corneil) [reply spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/ComposedTaskRunnerConfiguration.java:111]: The taskName doesn't change during that method?

Comment (corneil) [reply spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/ComposedTaskRunnerConfiguration.java:111]: Doesn;t names like foo-a and fooA clash in other areas?

Comment (cppwfs) [reply spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/ComposedTaskRunnerConfiguration.java:111]: Only if we process the whole property ID via the RelaxedNames will the foo-a get updated.

**Review by onobc — APPROVED** [review]
LGTM - thanks @cppwfs

**Comment by klopfdreh** [comment]
Hey @onobc and @cppwfs - I don't know the workflow, but will this also be merged into [main-3](https://github.com/spring-cloud/spring-cloud-dataflow/tree/main-3)?

**Comment by cppwfs** [comment]
Hello @klopfdreh Thank you for replying .  We are making sure all changes that apply to main-3 will be applied.

**Comment by klopfdreh** [comment]
Hey @cppwfs - thanks a lot!

**Pull request merged**